core/services/headtracker/head_tracker.go

```diff
@@ -274,28 +270,24 @@ func (ht *HeadTracker) Backfill(ctx context.Context, headWithChain eth.Head, dep
 }
 
 // backfill fetches all missing heads up until the base height
-func (ht *HeadTracker) backfill(ctxParent context.Context, head eth.Head, baseHeight int64) (err error) {
+func (ht *HeadTracker) backfill(ctxParent context.Context, head *eth.Head, baseHeight int64) (err error) {
 	if head.Number <= baseHeight {
 		return nil
 	}
 	mark := time.Now()
 	fetched := 0
-	ht.log.Debugw("Starting backfill",
-		"blockNumber", head.Number,
+	l := ht.log.With("blockNumber", head.Number,
 		"n", head.Number-baseHeight,
 		"fromBlockHeight", baseHeight,
 		"toBlockHeight", head.Number-1)
+	l.Debug("Starting backfill")
 	defer func() {
 		if ctxParent.Err() != nil {
 			return
 		}
-		ht.log.Debugw("Finished backfill",
+		l.Debugw("Finished backfill",
```

Comment:
This log line was accessing `head` which is modified below. In the old code, it would be a zeroed eth.Head in the event of error. After this change it became a nil pointer exposing the problem, and so now the fields are stored on the Logger instead.